src/problem.jl

```diff
@@ -17,14 +17,30 @@ function DiffEqJump.JumpProblem(prob,aggregator,rn::DiffEqBase.AbstractReactionN
     # get a JumpSet of the possible jumps
     jset = network_to_jumpset(rn, spec_to_idx, param_to_idx, prob.p)
 
-    # # construct dependency graph if needed by aggregator
+    # construct map from species index to indices of reactions that depend on it
+    if needs_vartojumps_map(aggregator) || needs_depgraph(aggregator)
```

Comment:
would it be a good idea to precompute some of this? This is re-computed each time solve is called. I don't know if that's a big deal though.

Reply:
Just to clarify; this is called each time a `JumpProblem` is made, not when `solve` is called right? We definitely don't want to redo this every time `solve` is called; it could be very expensive on larger networks. For `JumpProblem`s it would probably make sense to have optional kwargs that pass in the needed maps / dependency graphs, and only generate them if they aren't passed in, but I imagine only more advanced users with make use of that since one has to understand what is required for each specific SSA and how to get it from these routines. I'll can update to allow that possibility.

Reply:
I'm going to leave this for a future PR. We might want at some point to make a struct to encapsulate various dependency info, rather than passing around lots of different things. In that case the user could sent that into the `JumpProblem` if they already have it.